src/cmd/api/controllers/videos_list_handler.go

```diff
@@ -28,62 +31,10 @@ type VideoListResponse struct {
 }
 
 type VideosListHandler struct {
+	S3Client  clients.IS3Client
```

Comment:
It seems we don't need the S3Client here.
And then we don't need it too in the router